content/storage-history/history.textile

```diff
@@ -0,0 +1,596 @@
+    PaginatedResult<Message> resultPage = await channel.HistoryAsync();
+    Message recentMessage = resultPage.Items[0];
+    Console.WriteLine("Most recent message: " + recentMessage.Id + " - " + recentMessage.Data);
+    ```
+
+    ```[objc]
+    ARTRest *rest = [[ARTRest alloc] initWithKey:@"{{API_KEY}}"];
+    ARTRestChannel *channel = [rest.channels get:@"{{RANDOM_CHANNEL_NAME}}"];
+    [channel publish:@"example" data:@"message data"];
+    [channel history:^(ARTPaginatedResult<ARTMessage *> *resultPage, ARTErrorInfo *error) {
+        ARTMessage *recentMessage = resultPage.items[0];
+        NSLog(@"Most recent message: %@ - %@", recentMessage.id, recentMessage.data);
+    }];
+    ```
+
+    ```[swift]
+    let rest = ARTRest(key: "{{API_KEY}}")
+    let channel = rest.channels.get("{{RANDOM_CHANNEL_NAME}}")
+    channel.publish("example", data: "message data")
+    channel.history { resultPage, error in
+        let recentMessage = resultPage!.items[0] as! ARTMessage
+        print("Most recent message: \(recentMessage.id) - \(recentMessage.data)")
+    }
+    ```
+
+    ```[go]
+    rest, err := ably.NewRestClient(ably.NewClientOptions("{{API_KEY}}"))
+    channel := rest.Channels.Get("{{RANDOM_CHANNEL_NAME}}")
+    err = channel.Publish("example", "message data")
+    page, err := channel.History(nil)
+    fmt.Println("Last published message: %s\n", page.Messages[0].Data)
+
+
+<span lang="ruby">Note that all examples on this page assume you are running them within an EventMachine reactor. Find out more in our "realtime usage documentation":/LINK-Me.</span>
+
+h4(#channel-parameters). Channel history parameters
+
+The following query parameters can be included in a call to @history()@. Note that @untilAttach@ is only available when using the realtime interface of an Ably SDK:
+
+|_. Parameter |_. Description |
+| options | ensures that clients disconnected abruptly from Ably can resume their connection |
+| start | earliest time in milliseconds since the epoch for any messages retrieved |
+| end | latest time in milliseconds since the epoch for any messages retrieved |
+| direction | @forwards@ or @backwards@ |
+| limit | maximum number of messages to retrieve up to 1,000 |
+| untilAttach | when true, ensures message history is up until the point of the channel being attached. See "continuous history":#continuous-history for more info. Requires the @direction@ to be @backwards@ (the default). If the @channel@ is not attached, or if @direction@ is set to @forwards@, this option will result in an error. Realtime only. |
+
+h2(#presence-history). Retrieve presence history
+
+You can retrieve presence history by using the "@Presence@":/api/realtime-sdk/presence object to expose the "@history()@":/api/realtime-sdk/presence#history method, enabling a client to retrieve historical presence events on the channel.
+
+The following example retrieves the first two pages of historical presence events published:
```

Comment:
This is paginated, so it will continue to page through all results (rather than stopping at only 2).